src/scripts/modules/add_destination.js

```diff
@@ -27,53 +23,50 @@ function cacheDom() {
 
 function bindEvents() {
   form.addEventListener("submit", (event) => {
-    setValues();
     addDestination();
     event.preventDefault();
-    const itemIndex = destinationsList.length;
+    //get the index of the destination pushed to the destinationsList
+    const itemIndex = destinationsList.length - 1;
     events.publish("destinationAdded", itemIndex);
     console.log(destinationsList);
   });
 
   radioVisited.addEventListener("click", isVisited);
   radioBucketlist.addEventListener("click", isBucketlist);
+  // update the city
   cityInput.addEventListener("change", (event) => {
-    updateCity("city", event.target.value);
+    updateFormState("city", event.target.value);
   });
+  // ...the country
   countryInput.addEventListener("change", (event) => {
-    updateCountry("country", event.target.value);
+    updateFormState("country", event.target.value);
   });
+  //..and the photo of the new list item
   photoInput.addEventListener("change", (event) => {
-    updatePhoto("photo", event.target.value);
+    updateFormState("photo", event.target.value);
   });
 }
 
-function setValues() {
-  return {
-    city: cityInput,
-    country: countryInput,
-    visited: "",
-    url: photoInput,
-  };
-}
+let formState = {
```

Comment:
Although it works because of [hoisting], this should be initiated at the top of the function for better readability :)